File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/exceptions/InvalidMessageTypeException.java
package com.evernym.verity.sdk.exceptions;

/**
 * Exception class used express invalid message types when parsing incoming messages
 */
public class InvalidMessageTypeException extends VerityException {
    /**
     * Constructs a new InvalidMessageTypeException with the given message.
     *
     * @param message the given message for the cause of the exception.
     *
     * @see VerityException#VerityException(String)
     */
    public InvalidMessageTypeException(String message) {super(message);}

    /**
     * Constructs a new InvalidMessageTypeException with the given message and cause.
     *
     * @param message the given message for the cause of the exception.
     * @param cause the given cause (Throwable) for the exception.
     * 
     * @see VerityException#VerityException(String, Throwable)
     */
    public InvalidMessageTypeException(String message, Throwable cause) {super(message, cause);}
}

File: sdk/dotnet-sdk/src/VeritySDK/Exception/InvalidMessageTypeException.cs
using System;

namespace VeritySDK
{
    /// <summary>
    /// Exception class used express invalid message types when parsing incoming messages
    /// </summary>
    public class InvalidMessageTypeException : VerityException
    {
        /// <summary>
        /// Constructs a new exception with the given message.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        public InvalidMessageTypeException(string message) : base(message) { }

        /// <summary>
        /// Constructs a new exception with the given message and cause.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        /// <param name="cause">the given cause for the exception.</param>
        public InvalidMessageTypeException(String message, System.Exception cause) : base(message, cause) { }
    }
}
